src/Core/Configurations/RuntimeConfigValidator.cs

```diff
@@ -740,6 +740,7 @@ public void ValidateRelationshipsInConfig(RuntimeConfig runtimeConfig, ISqlMetad
         [System.Diagnostics.CodeAnalysis.SuppressMessage("Performance", "CA1822:Mark members as static", Justification = "Current use by dab workflow")]
         public void ValidateStoredProceduresInConfig(RuntimeConfig runtimeConfig, ISqlMetadataProvider sqlMetadataProvider)
         {
+            RequestValidator requestValidator = new(sqlMetadataProvider, _runtimeConfigProvider);
```

Comment:
Correct. But speaking of runtimeconfig validation, I think it should not have a dependency on request validator, as we are validating the config and not validating an actual request. However, it so happens that the method we have is present in RequestValidator class. And so, rather than creating a service dependency I initialized an instance of RequestValidator here. 

Another option I have is to make the function static, so we don't have to create this instance. What is your opinion about this?

Reply:
I'm unsure about why we are using a request validator to validate stored procedure configuration. @abhishekkumams , is this because you didn't want duplicate code?

Reply:
RequestValidator.ValidateStoredProcedureRequestContext checks for missing parameters, which is one of the checks that is required to do while settingUp itself, if fails it should throw error pointing out incorrect config file, that is why it is being called here. This was initially used by rest request but this is valid for GraphQL as well, so to not have duplication as pointed out by @seantleonard

Reply:
I think we're going to want to identify a way to not have the RequestValidator validating configuration. not in this pr

Reply:
Created an issue here: https://github.com/Azure/data-api-builder/issues/1717